watertap/core/tests/test_zero_order_costing.py

```diff
@@ -0,0 +1,128 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Tests for general zero-order costing methods
+"""
+import pytest
+
+from pyomo.environ import (
+    Block, ConcreteModel, Constraint, Var, check_optimal_termination, value)
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+from idaes.generic_models.costing import UnitModelCostingBlock
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from watertap.core.zero_order_costing import \
+    ZeroOrderCosting, ZeroOrderCostingData
+from watertap.core.zero_order_properties import WaterParameterBlock
+from watertap.core.wt_database import Database
+from watertap.unit_models.zero_order import NanofiltrationZO
+
+solver = get_solver()
+
+
+class TestWorkflow:
+    @pytest.fixture(scope="class")
+    def model(self):
+        m = ConcreteModel()
+        m.db = Database()
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.params = WaterParameterBlock(
+            default={"solute_list": ["sulfur", "toc", "tss"]})
+
+        return m
+
+    @pytest.mark.component
+    def test_nf_costing(self, model):
+        model.fs.unit = NanofiltrationZO(default={
+            "property_package": model.fs.params,
+            "database": model.db})
+
+        model.fs.unit.inlet.flow_mass_comp[0, "H2O"].fix(10000)
+        model.fs.unit.inlet.flow_mass_comp[0, "sulfur"].fix(1)
+        model.fs.unit.inlet.flow_mass_comp[0, "toc"].fix(2)
+        model.fs.unit.inlet.flow_mass_comp[0, "tss"].fix(3)
+        model.fs.unit.load_parameters_from_database()
+        assert degrees_of_freedom(model.fs.unit) == 0
+
+        model.fs.costing = ZeroOrderCosting()
```

Comment:
Why do we need to make an instance of ZeroOrderCosting and then replace with UnitModelCostingBlock()?

Reply:
We don't. Note the fist one is attached to `m.fs`, and the second one to `m.fs.unit`.

Reply:
Oops- you didn't replace anything. This is on flowsheet level and the next is on the unit level.